src/annotator/anchoring/pdf.js

```diff
@@ -541,12 +549,10 @@ export async function anchor(root, selectors) {
     // If anchoring with the position failed, check for a cached quote-based
     // match using the quote + position as a cache key.
     try {
-      if (
-        quotePositionCache[quote.exact] &&
-        quotePositionCache[quote.exact][position.start]
-      ) {
-        const { pageIndex, anchor } =
-          quotePositionCache[quote.exact][position.start];
+      const cacheKey = quotePositionCacheKey(quote.exact, position.start);
```

Comment:
On a future PR, I am a proponent of checking the cache first, by moving this try/catch above the previous one.

Reply:
Yes, this would make sense. It would be faster as it would avoid work to recompute the page index. Currently only quote-based anchoring populates this cache, but it would make sense to populate it for position-based anchoring too.